thoth/storages/graph/postgres.py

```diff
@@ -4677,6 +4695,106 @@ def _get_or_create_python_package_index(
+            )
+            HasMetadataPlatform.get_or_create(
+                self._session,
+                python_package_metadata_platform_id=python_package_metadata_platform.id,
+                python_package_metadata_id=package_metadata.id,
+            )
+
+        for supported_platform in importlib_metadata.pop("Supported-Platform", []):
+            python_package_metadata_supported_platform, _ = PythonPackageMetadataSupportedPlatform.get_or_create(
+                self._session,
+                supported_platform=supported_platform,
+            )
+            HasMetadataSupportedPlatform.get_or_create(
+                self._session,
+                python_package_metadata_supported_platform_id=python_package_metadata_supported_platform.id,
+                python_package_metadata_id=package_metadata.id,
+            )
+
+        for dependency in importlib_metadata.pop("Requires-External", []):
+            python_package_metadata_requires_external, _ = PythonPackageMetadataRequiresExternal.get_or_create(
+                self._session,
+                dependency=dependency,
+            )
+            HasMetadataRequiresExternal.get_or_create(
+                self._session,
+                python_package_metadata_requires_external_id=python_package_metadata_requires_external.id,
+                python_package_metadata_id=package_metadata.id,
+            )
+
+        for project_url in importlib_metadata.pop("Project-URL", []):
+            python_package_metadata_project_url, _ = PythonPackageMetadataProjectUrl.get_or_create(
+                self._session,
+                project_url=project_url,
+            )
+            HasMetadataProjectUrl.get_or_create(
+                self._session,
+                python_package_metadata_project_url_id=python_package_metadata_project_url.id,
+                python_package_metadata_id=package_metadata.id,
+            )
+
+        for project_url in importlib_metadata.pop("Provides-Extra", []):
+            python_package_metadata_provides_extra, _ = PythonPackageMetadataProvidesExtra.get_or_create(
+                self._session,
+                project_url=project_url,
+            )
+            HasMetadataProvidesExtra.get_or_create(
+                self._session,
+                python_package_metadata_provides_extra_id=python_package_metadata_provides_extra.id,
+                python_package_metadata_id=package_metadata.id,
+            )
+
+        for dist_key in ["Requires-Dist", "Provides-Dist", "Obsoletes-Dist"]:
+            if "Requires-Dist":
+                distutils_type = MetadataDistutilsTypeEnum.REQUIRED.value
+
+            elif "Provides-Dist":
+                distutils_type = MetadataDistutilsTypeEnum.PROVIDED.value
+
+            elif "Obsoletes-Dist":
+                distutils_type = MetadataDistutilsTypeEnum.OBSOLETE.value
+
+            else:
+                _LOGGER.warning("No distutils key registered for %r", dist_key)
```

Comment:
This will cause the `distutils_type` not being defined on the model created bellow causing a crash during execution. You can directly raise an exception here.

Reply:
I'm gonna create a specific exception then, thanks :)